packages/regenerator-transform/src/hoist.js

```diff
@@ -87,7 +87,7 @@ exports.hoist = function(funPath) {
           "=",
           node.id,
           t.functionExpression(
-            node.id,
+            null,
```

Comment:
Can you use `path.scope.generateUidIdentifierBasedOnNode(node)`? By doing this, we give a name to the function (so it helps during debugging) and we avoid shadowing the original variable.